Source/Data/SpectrumAnalysis/KTSeqLine.cc

```diff
@@ -0,0 +1,273 @@
+                    //Calculating error on the starting frequency and the end frequency
+                    double startTime = *std::min_element(timeBinInAcq.begin(), timeBinInAcq.end());
+                    double endTime = *std::max_element(timeBinInAcq.begin(), timeBinInAcq.end());
+                    sigmaStartFreq = sqrt( startTime*startTime *  deltaSlope*deltaSlope + deltaIntercept*deltaIntercept + 2 * startTime * rho * deltaSlope * deltaIntercept );
+                    sigmaEndFreq = sqrt( endTime*endTime *  deltaSlope*deltaSlope + deltaIntercept*deltaIntercept + 2 * endTime * rho * deltaSlope * deltaIntercept );
+                }
+
+                // TODO: Calculate distance to track and see for a possible alpha [%] rejection of noise.
+
+                //fSlope = slope;
+                fIntercept = intercept;
+                fSlopeSigma = deltaSlope;
+                fInterceptSigma = deltaIntercept;
+                fStartFrequencySigma = sigmaStartFreq;
+                fEndFrequencySigma = sigmaEndFreq;
+
+    }
+
+    inline void LineRef::CalculateSlope()
+    {
+
+        //KTDEBUG(seqlog, "Calculating line slope");
+        double weightedSlope = 0.0;
+        double wSum = 0.0;
+
+        if (fNPoints > 1)
+        {
+            for(std::vector<LinePoint>::iterator PointIt = fLinePoints.begin(); PointIt != fLinePoints.end(); ++PointIt)
+            {
+                if (PointIt->fPointFreq > fStartFrequency)
+                {
+                    weightedSlope += (PointIt->fPointFreq - fStartFrequency)/(PointIt->fTimeInAcq - fStartTimeInAcq) * PointIt->fAmplitude;
+                    wSum += PointIt->fAmplitude;
+                }
+            }
+            fSlope = weightedSlope/wSum;
+        }
+        else
+        {
+            fSlope = fInitialSlope;
+        }
+    }
+
+    void LineRef::LineTrimming(const double& TrimmingFactor, const unsigned& MinPoints)
+    {
+        KTDEBUG(seqlog, "Trimming line edges");
+
+        if (!fAmplitudeList.empty())
+        {
+            while (fAmplitudeList.back() < TrimmingFactor * fTrimmingLimits.back() and fNPoints >= MinPoints)
+            {
+                fAmplitudeList.erase(fAmplitudeList.end() -1);
+                fTrimmingLimits.erase(fTrimmingLimits.end() -1);
+                fLinePoints.erase(fLinePoints.end() - 1);
+                fNPoints = fLinePoints.size();
+            }
+            while (fAmplitudeList.front() < TrimmingFactor * fTrimmingLimits.front() and fNPoints >= MinPoints)
+            {
+                fAmplitudeList.erase(fAmplitudeList.begin());
+                fTrimmingLimits.erase(fTrimmingLimits.begin());
+                fLinePoints.erase(fLinePoints.begin());
+                fNPoints = fLinePoints.size();
+            }
+        }        
+        this->UpdateLineParameters();
+        this->CalculateSlope();
+    }
+        
+
+    inline void (LineRef::UpdateLineParameters())
```

Comment:
This function isn't inline, and what's with the extra parentheses around the function name?

Variable name to fix: `PointIt`

Also `++` should be prefixed on `pointIt`